File: Exercism/typescript/leap/leap.ts
export default function isLeapYear(year: number): boolean {
    return specialCaseEveryOneHundredYears(year)
        ? everyFourHundredYearsIgnoreSpecialCase(year)
        : shouldBeLeapYear(year)
}

function specialCaseEveryOneHundredYears(year: number) {
    return year % 100 === 0
}
function everyFourHundredYearsIgnoreSpecialCase(year: number) {
    return year % 400 === 0
}
function shouldBeLeapYear(year: number) {
    return year % 4 === 0
}

File: Exercism/csharp/leap/Leap.cs
using System;

public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        return SpecialCaseEveryOneHundredYears(year)
            ? EveryFourHundredYearsIgnoreSpecialCase(year)
            : ShouldBeLeapYear(year);
    }

    public static bool SpecialCaseEveryOneHundredYears(int year)
    {
        return year % 100 == 0;
    }

    public static bool EveryFourHundredYearsIgnoreSpecialCase(int year)
    {
        return year % 400 == 0;
    }

    public static bool ShouldBeLeapYear(int year)
    {
        return year % 4 == 0;
    }
}